dashboard/src/shared/ResourceRef.ts

```diff
@@ -30,19 +30,29 @@ export function fromCRD(
 class ResourceRef {
   public cluster: string;
   public apiVersion: string;
-  public kind: ResourceKind;
+  public kind: string;
+  public plural: string;
+  public namespaced: boolean;
   public name: string;
   public namespace: string;
   public filter: any;
 
   // Creates a new ResourceRef instance from an existing IResource. Provide
   // defaultNamespace to set if the IResource doesn't specify a namespace.
-  constructor(r: IResource, cluster: string, defaultNamespace?: string) {
+  constructor(
+    r: IResource,
+    cluster: string,
+    plural: string,
+    namespaced: boolean,
+    defaultNamespace?: string,
+  ) {
```

Comment:
I think this is the main change here - adding the extra data (plural, namespaced) to the constructor so that it can be used. What's not clear to me is why we need to use `plural` at all. I guess I'll wait for the non-draft PR with an explanation :) Thanks!

Reply:
Yes. The `plural` is needed to build the URL from the `kind` (which is the only information that we have for sure). So we can resolve, for example, `kind: Role` -> `/apis/rbac.authorization.k8s.io/v1/namespaces/kubeapps/roles/<name>`

This was hardcoded at https://github.com/kubeapps/kubeapps/pull/2234/files/6f856ca662950a5d97f960825b23af58458eb761#diff-f6be03870e629fd9c812827f468e126171394db8ac54eea771d90cd6d8c0925a

Reply:
https://github.com/kubernetes/kubernetes/issues/18622

In particular:

```
The current impl (changed in November, I think) now pluralizes ListOptions as listoptionses, PodLogOptions as podlogoptionses, etc. I'm as big a LotR fan as anyone, but...
```
:)

Reply:
:sweat_smile: it would have been nice to not have plurals... even though we'd still need to discover the apiVersions and if the resources are namespaced or not.